### PR TITLE
Add Firefly v2 brigthness support

### PR DESCRIPTION
**Pull request by paxos**
Add initial support for brightness for the Firefly V2 mousemat

Screenshot 2021-04-12 at 22 56 30
Screenshot 2021-04-12 at 22 57 03

Fixes https://github.com/1kc/razer-macos/issues/352

### CODE REVIEW

File: src/main/menu/menubuilderdevice.js [hunk 1]
```diff
@@ -111,7 +113,7 @@ function getFeatureBrightness(application, device, feature) {
     label: 'Brightness',
     submenu: [
       {
-        label: `Brightness: ${device.brightness}%`,
+        label: `Brightness: ${device.settings.customBrightness}%`,
```
Comment (paxos) [inline src/main/menu/menubuilderdevice.js:116]: Unclear if this breaks existing feature or if it worked before?

**Comment by dhobi** [comment]
Hi, thanks for the PR! 👍

I have quickly reviewed your changes and have noticed that your FeatureMouseMatBrightness is very similar to FeatureBrightness (so far for the keyboard only).

Proposal:
I think we could get rid of the extra feature class you added and adapt some code to reuse the already existing FeatureBrightness for mouse mats as well

- check the device.mainType for the labels in the gui. 
- The sent event in the feature brightness component could be renamed from 'update-keyboard-brightness' to 'update-brightness'

The rest looks good to me!

**Comment by paxos** [comment]
Sounds good 👍 Will post an update with an simpler version.

Both features are very similar! I copied a lot of things, mostly for clarity.

**Comment by paxos** [comment]
@dhobi I simplified the PR and removed duplicate code. Let me know if this one works 👍 (feel free to modify)

**Comment by dhobi** [comment]
Yes, the updated version looks cleaner to me. Thanks for updating it! 

The only thing which I have a bit of a blind spot is the device.settings.customBrightness property. Are we actually using (read) this somewhere or should we just get rid of it completely and use the device.brightness property instead? What do you think?

**Comment by paxos** [comment]
I am not 100% sure why we have both settings. I chose to use `customBrightness` as it was already used by the keyboard setter. Not sure if the menu ever worked there and I can not test it as I don't have a Razer keyboard.

I can run some tests to see if we can merge both properties. But this is probably going to take a while. Please feel free to take over and push changes if you have ideas for a quick fix 👍

**Comment by dhobi** [comment]
I've just updated the branch code with the following:
- added Goliathus products
- add brightness read function (even though most mousemats won't support it sadly...we just return 100 in this case)
- remove settings.customBrightness and use device.brightness everywhere
- add brightness property to device state (create / reset)

Tested with my Goliathus Extended. Could you check if it still works for you (FIREFLY_V2)?

Thanks for the PR once again!

**Comment by paxos** [comment]
@dhobi confirmed, everything working perfectly with `FIREFLY_V2`!

> Thanks for the PR once again!

Pleasure! Thanks so much for this project 🎉

**Review by dhobi — APPROVED** [review]
I think we can merge this then!

**Comment by 1kc** [comment]
Merged and released under v0.4.2!

**Pull request merged**